plugins/inventory/linode.py

```diff
@@ -326,3 +322,43 @@ def parse(self, inventory, loader, path, cache=True):
                 variables,
                 instance.label,
                 strict=strict)
+
+    def verify_file(self, path):
+        """Verify the Linode configuration file."""
+        if super(InventoryModule, self).verify_file(path):
+            endings = ('linode.yaml', 'linode.yml')
+            if any((path.endswith(ending) for ending in endings)):
+                return True
+        return False
+
+    def parse(self, inventory, loader, path, cache=True):
+        """Dynamically parse Linode the cloud inventory."""
+        super(InventoryModule, self).parse(inventory, loader, path)
+        self.instances = []
+
+        if not HAS_LINODE:
+            raise AnsibleError('the Linode dynamic inventory plugin requires linode_api4.')
+
+        config_data = self._read_config_data(path)
+        self._consume_options(config_data)
+
+        cache_key = self.get_cache_key(path)
+
+        if cache:
+            cache = self.get_option('cache')
+
+        update_cache = False
+        if cache:
+            try:
+                self.instances = [Instance(None, i["id"], i) for i in self._cache[cache_key]]
+            except KeyError:
+                update_cache = True
+
+        if not self.instances:
```

Comment:
So here you could do:
```suggestion
        if self.instances is not None:
```
to distinguish between "the cached set of instances was empty" and "nothing was cached".

Reply:
Good catch! That's clever. I will add this on Monday.

Reply:
Added 👍🏻